lib/sequent/test/command_handler_helpers.rb

```diff
@@ -84,17 +89,26 @@ def deserialize_events(events)
 
       end
 
-      def given_streams_with_events *streams_with_events
-        @event_store.given_events(streams_with_events)
+      def given_events(aggregate_class, aggregate_id, *events)
+        @event_store.given_events [Sequent::Core::EventStream.new(aggregate_type: aggregate_class.to_s, aggregate_id: aggregate_id, snapshot_threshold: 30, stream_record_id: 1), events]
       end
 
+
       def when_command command
         raise "@command_handler is mandatory when using the #{self.class}" unless @command_handler
         raise "Command handler #{@command_handler} cannot handle command #{command}, please configure the command type (forgot an include in the command class?)" unless @command_handler.handles_message?(command)
-        @command_handler.handle_message(command)
+        begin
+          @command_handler.handle_message(command)
+        rescue Sequent::Core::DomainError => e
+          @actual_error = e
+        end
         @repository.commit(command)
       end
 
+      def then_fails_with clazz
+        expect(@actual_error.class).to be(clazz)
+      end
+
       def then_events *events
```

Comment:
I would expect a command to either fail or result in new events, but not both. An error would result in a transaction rollback anyway.

As errors are now silently ignored on executing the command, it is helpful to raise them (with their original stack) as `then_events` is called. Otherwise tests may pass where the code actually raises an error.